pkg/autodiscovery/providers/kube_endpoints.go

```diff
@@ -30,9 +32,11 @@ const (
 
 // KubeEndpointsConfigProvider implements the ConfigProvider interface for the apiserver.
 type KubeEndpointsConfigProvider struct {
-	serviceLister   listersv1.ServiceLister
-	endpointsLister listersv1.EndpointsLister
-	upToDate        bool
+	serviceLister      listersv1.ServiceLister
+	endpointsLister    listersv1.EndpointsLister
+	upToDate           bool
+	monitoredEndpoints map[string]bool
+	m                  sync.RWMutex
```

Comment:
just a code style remark when dealing with Mutex that I tried to apply: if a Mutex is used to protect several fields in a struct, I put the Mutex as an EmbeddedField 
```
type KubeEndpointsConfigProvider struct {
	sync.RWMutex
}
```

And also since `NewKubeEndpointsConfigProvider()` return an interface the struct `KubeEndpointsConfigProvider ` can be private `kubeEndpointsConfigProvider `

Reply:
thanks 🙇